src/test/fuzz/compression_roundtrip.cpp

```diff
@@ -0,0 +1,442 @@
+#include <node/transaction.h>
+#include <node/chainstate.h>
+#include <node/miner.h>
+#include <node/coin.h>
+#include <primitives/transaction.h>
+#include <primitives/compression.h>
+#include <rpc/server_util.h>
+#include <rpc/util.h>
+#include <test/util/setup_common.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <util/strencodings.h>
+#include <cassert>
+#include <chainparams.h>
+#include <cmath>
+#include <coins.h>
+#include <core_io.h>
+#include <iostream>
+#include <key_io.h>
+#include <pow.h>
+#include <random.h>
+#include <secp256k1.h>
+#include <secp256k1_extrakeys.h>
+#include <streams.h>
+#include <uint256.h>
+#include <univalue.h>
+#include <validation.h>
+
+using node::BlockManager;
+using node::BlockAssembler;
+using node::CBlockTemplate;
+using node::GetTransaction;
+using node::RegenerateCommitments;
+using node::FindCoins;
+
+namespace {
+    class SecpContext {
+    		secp256k1_context* ctx;
+
+    	public:
+    		SecpContext() {
+    			ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
+    		}
+    		~SecpContext() {
+    			secp256k1_context_destroy(ctx);
+    		}
+    		secp256k1_context* GetContext() {
+    			return ctx;
+    		}
+    };
+    struct CompressionRoundtripFuzzTestingSetup : public TestChain100Setup {
+    	CompressionRoundtripFuzzTestingSetup(const ChainType& chain_type, const std::vector<const char*>& extra_args) : TestChain100Setup{chain_type, extra_args}
+    	{}
+
+        std::vector<unsigned char> GetSerializedPubKey(secp256k1_pubkey pubkey, bool compressed) {
+            size_t p_size = 33;
+            auto p_type = SECP256K1_EC_COMPRESSED;
+            if (!compressed) {
+                p_size = 65;
+                p_type = SECP256K1_EC_UNCOMPRESSED;
+            }
+            std::vector<unsigned char> p_vec(p_size);
+            secp256k1_ec_pubkey_serialize(secp256k1_context_static, &p_vec[0], &p_size, &pubkey, p_type);
+            return p_vec;
+        };
+
+        std::tuple<uint32_t, std::vector<CCompressedOutPoint>, std::vector<CScript>> CompressOutPoints(CTransaction tx) {
+            LOCK(cs_main);
+            auto tuple = EnsureChainman(m_node).ActiveChainstate().CompressOutPoints(tx);
+            return std::make_tuple(std::get<0>(tuple), std::get<1>(tuple), std::get<3>(tuple));
+        }
+
+        std::tuple<std::vector<COutPoint>, std::vector<CTxOut>> UncompressOutPoints(CCompressedTransaction tx) {
+            LOCK(cs_main);
+            std::vector<COutPoint> prevouts = EnsureChainman(m_node).ActiveChainstate().UncompressOutPoints(tx);
+            std::vector<Coin> coins = FindCoins(m_node, prevouts);
+            std::vector<CTxOut> outs;
+            for (auto& coin : coins) {
+                outs.push_back(coin.out);
+            }
+            return std::make_tuple(prevouts, outs);
+        }
+
+    	bool IsTopBlock(CBlock block) {
+    		LOCK(cs_main);
```

Comment:
In c6f4427153b02f4b66535c9b8fdb0e9a2efa35cc:

Lots of weird spacing in this file. Please go through all your commits and check for tabs.

Reply:
Fixed, Thanks!